src/lib/os.pl

```diff
@@ -139,7 +139,8 @@
     '$argv'(Argv0),
     ( member("--", Argv0) ->
         append(Argv1, ["--"|Argv], Argv0),
-        \+ member("--", Argv1)
+        \+ member("--", Argv1),
+        !
```

Comment:
If only the first solution is needed, you can use `once/1`: **`once(append(Argv1, ...))`**

Reply:
I wasn't aware of once.
Given order of solutions of append that should indeed work.
Added a test case to be sure and updated the code to use once instead of the cut.